packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -122,4 +142,28 @@ export class EditorPreviewManager extends EditorManager {
             widget.convertToNonPreview();
         }
     }
+
+    protected override async revealSelection(widget: EditorWidget, options: EditorOpenerOptions = {}, uri?: URI): Promise<void> {
+        if (!options.selection) {
+            options.selection = await this.resolveSelection(options, widget, uri);
+        }
+        super.revealSelection(widget, options, uri);
+    }
+
+    protected async resolveSelection(options: EditorOpenerOptions, widget: EditorWidget, uri: URI | undefined): Promise<RecursivePartial<Range> | undefined> {
+        if (!options.selection) {
+            const orderedResolvers = Prioritizeable.prioritizeAllSync(this.resolvers.getContributions(), resolver => resolver.priority ?? 1);
```

Comment:
If the set of resolvers is fixed at initialisation time, we can sort the resolvers once and keep the result. I assume the priority of a resolver is not supposed to ever change, right?